src/connect/hooks/tree-view/useItemActions/useItemActions.ts

```diff
@@ -0,0 +1,158 @@
+import { ActionType, TreeItem } from '@/connect/components/tree-view-item';
+import { useItemsContext, UITreeItemState } from '@/connect/context/ItemsContext';
+
+export const useItemActions = () => {
+    const { setUIState, setVirtualItems, baseItems } = useItemsContext();
+    
+    /**
+     * Sets the item as selected in the UI state.
+     * @param itemID - The ID of the item to be selected.
+     */
+    const setSelectedItem = (itemID: string) => {
+        setUIState((prevState) => {
+            const uiValues = Object
+                .entries(prevState)
+                .reduce<UITreeItemState>((acc, entry) => {
+                    const [key, value] = entry;
+                    const uiState = {
+                        ...acc,
+                        [key]: { ...value, isSelected: false },
+                    };
+
+                    return uiState;
+                }, {});
+
+            const newItemState = {
+                [itemID]: {
+                    ...prevState[itemID],
+                    isSelected: true,
+                },
+            };
+
+            return { ...uiValues, ...newItemState };
+        });
+    };
+
+    /**
+     * Sets the item as expanded in the UI state.
+     * @param itemID - The ID of the item.
+     * @param expanded - The expanded state to set (default: true).
+     */
+    const setExpandedItem = (itemID: string, expanded = true) => {
+        setUIState((prevState) => ({
+            ...prevState,
+            [itemID]: {
+                ...prevState[itemID],
+                expanded,
+            },
+        }));
+    };
+
+    /**
+     * Toggles the expanded state of an item and selects it.
+     * 
+     * @param itemID - The ID of the item to toggle and select.
+     */
+    const toggleExpandedAndSelect = (itemID: string) => {
+        const baseItemExpanded = baseItems.find((item) => item.id === itemID)?.expanded || false;
+
+        setUIState((prevState) => {
+            const uiValues = Object
+                .entries(prevState)
+                .reduce<UITreeItemState>((acc, entry) => {
+                    const [key, value] = entry;
+                    const uiState = {
+                        ...acc,
+                        [key]: {
+                            ...value,
+                            isSelected: false,
+                        },
+                    };
+
+                    return uiState;
+                }, {});
+
+
+            // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
```

Comment:
You don't need to disable this rule. we cant ever reach this point in the code without `prevState[itemID]` being defined, so the ? is not needed

Reply:
prevState contains the uiState, which is initially created with an empty value `{}`
when we try to access the previous state (for the first time) in this line for the `itemID`, there's no record in the uiState so the result of this code `prevState[itemID]` is `undefined`

Reply:
Then that means the type is telling the wrong story

```
// eslint-disable-next-line @typescript-eslint/consistent-indexed-object-style
export interface UITreeItemState {
    [key: string]: UITreeItem;
}
```

I still recommend using the `Record` type instead as this will automatically account for the fact that the value may be undefined when doing an index action. Otherwise, you'll need to do 

```/
/ eslint-disable-next-line @typescript-eslint/consistent-indexed-object-style
export interface UITreeItemState {
    [key: string]: UITreeItem | undefined;
}
```

or else other users of your code won't know that they have a potentially undefined value and will end up with runtime errors

Reply:
yes!, already fixed this type issue. I'll push a new commit with this change and some others 👍🏻

Reply:
fixed: [4b1a171](https://github.com/powerhouse-inc/design-system/pull/94/commits/4b1a171e9b9f348177a8710eb0113bfc32a737ad)